File: library/src/test/java/me/proxer/library/api/apps/ErrorLogEndpointTest.java
package me.proxer.library.api.apps;

import me.proxer.library.ProxerTest;
import me.proxer.library.api.ProxerException;
import okhttp3.mockwebserver.MockResponse;
import org.junit.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

public class ErrorLogEndpointTest extends ProxerTest {

    @Test
    public void testDefault() throws ProxerException, IOException {
        server.enqueue(new MockResponse().setBody(fromResource("empty.json")));

        final Void result = api.apps()
                .errorLog("3", "test message")
                .build()
                .execute();

        assertThat(result).isNull();
    }

    @Test
    public void testPath() throws ProxerException, IOException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("empty.json")));

        api.apps().errorLog("3", "test message")
                .build()
                .execute();

        assertThat(server.takeRequest().getPath()).isEqualTo("/api/v1/apps/errorlog");
    }

    @Test
    public void testParameters() throws IOException, ProxerException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("empty.json")));

        api.apps().errorLog("3", "test message")
                .anonym(false)
                .build()
                .execute();

        assertThat(server.takeRequest().getBody().readUtf8()).isEqualTo("id=3&message=test%20message&anonym=false");
    }
}

File: library/src/test/kotlin/me/proxer/library/api/apps/ErrorLogEndpointTest.kt
package me.proxer.library.api.apps

import me.proxer.library.ProxerTest
import me.proxer.library.runRequest
import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldBeNull
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class ErrorLogEndpointTest : ProxerTest() {

    @Test
    fun testDefault() {
        val (result, _) = server.runRequest("empty.json") {
            api.apps
                .errorLog("3", "test message")
                .build()
                .execute()
        }

        result.shouldBeNull()
    }

    @Test
    fun testPath() {
        val (_, request) = server.runRequest("empty.json") {
            api.apps.errorLog("3", "test message")
                .build()
                .execute()
        }

        request.path shouldBeEqualTo "/api/v1/apps/errorlog"
    }

    @Test
    fun testParameters() {
        val (_, request) = server.runRequest("empty.json") {
            api.apps.errorLog("3", "test message")
                .anonym(false)
                .silent(false)
                .build()
                .execute()
        }

        request.body.readUtf8() shouldBeEqualTo "id=3&message=test%20message&anonym=false&silent=false"
    }
}
